malfeasance/handler_test.go

```diff
@@ -656,7 +689,7 @@ func TestHandler_HandleMalfeasanceProof_hareEquivocation(t *testing.T) {
 		}
 		data, err := codec.Encode(gossip)
 		require.NoError(t, err)
-		require.Error(t, h.HandleMalfeasanceProof(context.Background(), "peer", data))
+		require.ErrorIs(t, h.HandleMalfeasanceProof(context.Background(), "peer", data), pubsub.ErrValidationReject)
```

Comment:
afaik this will drop the peer that sent the proof. If there are existing proofs of that kind did we update the code to not broadcast / serve them any more?

Reply:
OK I think I have misunderstood the test. I read the test as "hare eligibility proofs are deprecated and we drop peers that broadcast them" instead of only if those proofs are invalid.

Existing (valid) hare eligibilities proof do not cause peers to be dropped? If they are deprecated do we still need to keep and broadcast them indefinitely?

Reply:
The logic around handling proofs with `Eligibility` field set was not changed in this PR. Check: https://github.com/spacemeshos/go-spacemesh/blob/df074909b5b4c85328c5ae91581bd082ade9062d/malfeasance/handler.go#L120-L126